pkg/operator/ceph/csi/util.go

```diff
@@ -84,6 +84,20 @@ func templateToDeployment(name, templateData string, p templateParam) (*apps.Dep
 	return &dep, nil
 }
 
+func templateTologrotateSidecar(name, templateData string, p templateParam) (corev1.Container, error) {
```

Comment:
How about if this method applies all the needed artifacts to the pod spec? For example, it would add the script, the volumes, volumeMounts, and arguments. Then there is no need to touch the csi templates in so many places, and we can have this helper method completely implement it in a single place. I believe this is also what Blaine was referring to [here](https://github.com/rook/rook/pull/14305#pullrequestreview-2113482698).
```suggestion
func applyLogrotateSidecar(specTemplate *corev1.PodSpecTemplate, name, templateData string, p templateParam) corev1.Container {
```

Reply:
Done